libraries/hotstuff/qc_chain.cpp

```diff
@@ -498,19 +489,21 @@ namespace eosio { namespace hotstuff {
          return;
       }
 
-      bool quorum_met = _current_qc.quorum_met; //check if quorum already met
+      bool quorum_met = _current_qc.is_quorum_met(); //check if quorum already met
 
       // If quorum is already met, we don't need to do anything else. Otherwise, we aggregate the signature.
       if (!quorum_met){
 
          auto increment_version = fc::make_scoped_exit([this]() { ++_state_version; });
 
-         if (_current_qc.active_finalizers>0)
-            _current_qc.active_agg_sig = fc::crypto::blslib::aggregate({_current_qc.active_agg_sig, vote.sig });
+         boost::dynamic_bitset finalizer_set = _current_qc.get_active_finalizers();
+         if (finalizer_set.any())
+            _current_qc.set_active_agg_sig(fc::crypto::blslib::aggregate({_current_qc.get_active_agg_sig(), vote.sig }));
          else
-            _current_qc.active_agg_sig = vote.sig;
+            _current_qc.set_active_agg_sig(vote.sig);
```

Comment:
I don't think this is ideal. I think we should have a `bls_signature& operator+=(const bls_signature&)` and replace all this with:
```
    _current_qc.active_agg_sig += vote.sig;
```

Reply:
I would be ok with an `aggregate` method on `bls_signature`, but I don't like `operator+=`.

Reply:
`operator+=()` seems like a very natural fit to me, but I'm fine with `aggregate`.